LEGO1/mxtransitionmanager.h

```diff
@@ -2,19 +2,58 @@
 #define MXTRANSITIONMANAGER_H
 
 #include "mxcore.h"
+#include "mxvideopresenter.h"
+#include "legovideomanager.h"
+#include "legoomni.h"
 
-class MxVideoPresenter;
 
-// 0x100d7ea0
+// VTABLE 0x100d7ea0
 class MxTransitionManager : public MxCore
 {
 public:
-  MxTransitionManager();
-  virtual ~MxTransitionManager() override; // vtable+0x0
+    MxTransitionManager();
+    virtual ~MxTransitionManager() override; // vtable+0x0
 
-  __declspec(dllexport) void SetWaitIndicator(MxVideoPresenter *videoPresenter);
+    __declspec(dllexport) void SetWaitIndicator(MxVideoPresenter *videoPresenter);
 
-  virtual MxResult Tickle(); // vtable+0x8
+    virtual MxResult Tickle(); // vtable+0x8
+
+    // OFFSET: LEGO1 0x1004b950
+    inline virtual const char *ClassName() const override // vtable+0x0c
+    {
+        return "MxTransitionManager";
+    }
+
+    // OFFSET: LEGO1 0x1004b960
+    inline virtual MxBool IsA(const char *name) const override // vtable+0x10
+    {
+        return !strcmp(name, MxTransitionManager::ClassName()) || MxCore::IsA(name);
+    }
+
+    virtual MxResult GetDDrawSurfaceFromVideoManager(); // vtable+0x14
+
+    enum TransitionType {
+        NOT_TRANSITIONING,
+        NO_ANIMATION,
+        DISSOLVE,
+        PIXELATION,
+        SCREEN_WIPE,
+        WINDOWS,
+        BROKEN // Unknown what this is supposed to be, it locks the game up
+    };
+    
+    MxResult StartTransition(TransitionType p_animationType, MxS32 p_speed, MxU8 p_unk, MxBool p_playMusicInAnim);
```

Comment:
Similarly here, using `undefined` for `p_unk` would be cleaner (unless you know more than the code shows)